libraries/lib-network-manager/curl/CurlHandleManager.cpp

```diff
@@ -0,0 +1,349 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file CurlHandleManager.cpp
+ @brief Define a class responsible for reuse of CURL hanldes.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "CurlHandleManager.h"
+
+#include <algorithm>
+#include <sstream>
+
+#include <wx/platinfo.h>
+
+namespace audacity
+{
+namespace network_manager
+{
+namespace
+{
+
+void GetOSString (std::ostringstream& output, const wxPlatformInfo& platformInfo)
+{
+    const wxOperatingSystemId osID = platformInfo.GetOperatingSystemId ();
+
+    if (osID & wxOS_WINDOWS)
+        output << "Windows ";
+    else if (osID & wxOS_MAC)
+        output << "MacOS ";
+    else if (osID & wxOS_UNIX_LINUX)
+		output << "Linux ";
+	else if (osID & wxOS_UNIX_FREEBSD)
+		output << "FreeBSD ";
+	else if (osID & wxOS_UNIX_OPENBSD)
+		output << "OpenBSD ";
+	else
+		output << "Other ";
+
+	output << 
+        platformInfo.GetOSMajorVersion () <<
+		"_" <<
+		platformInfo.GetOSMinorVersion () <<
+        "_" <<
+        platformInfo.GetOSMicroVersion() << 
+		"; ";
+
+#if defined(__amd64__) || defined(__x86_64__) || defined(_M_X64)
+    output << "x64";
+#elif defined(__i386__) || defined(i386) || defined(_M_IX86) || defined(_X86_) || defined(__THW_INTEL)
+    output << "x86";
+#elif 
```

Comment:
This breaks the build on ARM: `error: #elif with no expression`

Reply:
That is now fixed, thank you